### DISCUSSION TITLE
dde-control-center 插件不显示问题

### DISCUSSION THREAD

**Discussion by jiangzc** [linuxdeepin/developer-center#2869] [Q&A | 问答板块] [answered]
你好，我是UOS社区开发者。

最近看到DTK可以开发控制中心的插件，我按照文档 https://docs.deepin.org/pages/7b0096/  的内容编译了源码中的 privacy 插件并安装到了指定目录。修改 gsetting 后重新运行控制中心，但没有看到对应的插件内容。

我的开发环境是：虚拟机  UOS 专业版 1031 ，编译环境使用的是UOS源中对应的开发库

操作步骤：
1. 下载 github  dde-control-center  uos分支最新的源代码
2. 运行 dpkg-checkbuilddeps 查看依赖，并用 apt 安装相关的开发库
3. 进入 privacy 目录，编译得到 libprivacy.so 安装到指定位置
4. 修改 gsetting 相关配置
5. 重新运行 dde-control-center

按照以上的操作步骤，我没有看到控制中心的插件内容。所以想请教一下dde-control-center 插件问题，期待您的回复，谢谢。

**Comment by shaotianlu1995** [comment, marked as answer]
你好，我是DDE的开发，应该可以帮助到你。

1、在.pro同级目录新建MetaData.json文件
填充内容为 `{"api":"1.0.0"}`

2、需要修改程序
修改privacyplugin.h文件的第55行
 `    Q_PLUGIN_METADATA(IID ModuleInterface_iid FILE "MetaData.json")`

3、重新编译程序

4、拷贝.so文件到 /usr/lib/dde-control-center/modules/中

一通操作之后，应该可以解决你的问题，感谢你对UOS的支持。

**Comment by LuciusV** [comment]
> 你好，我是DDE的开发，应该可以帮助到你。
> 
> 1、在.pro同级目录新建MetaData.json文件
> 填充内容为 `{"api":"1.0.0"}`
> 
> 2、需要修改程序
> 修改privacyplugin.h文件的第55行
> ` Q_PLUGIN_METADATA(IID ModuleInterface_iid FILE "MetaData.json")`
> 
> 3、重新编译程序
> 
> 4、拷贝.so文件到 /usr/lib/dde-control-center/modules/中
> 
> 一通操作之后，应该可以解决你的问题，感谢你对UOS的支持。

您好：
请注意https://github.com/linuxdeepin/dde-control-center/issues/363

**Reply by BLumia** [reply]
您所链接的 issue 和此主题无关，建议您在相关的主题中跟进问题。（另注：近期在处理之前未及时响应的陈旧 issue，您所链接的问题已迁移到了[这里](https://github.com/linuxdeepin/developer-center/issues/2749)）

**Reply by BLumia** [reply]
顺便，您可以考虑在 https://github.com/linuxdeepin/developer-center/discussions/2870 讨论下你的想法 :)
